README.md

```diff
@@ -112,6 +112,43 @@ title,content
 My Title,Hello World!
 ```
 
+### Throttling
+
+Maintenance Tasks often modify a lot of data and can be taxing on your database.
+The gem provides a throttling mechanism that can be used to throttle a Task when
+a given condition is met. Specify the throttle condition as a Proc:
+
+```ruby
+# app/tasks/maintenance/update_posts_throttled_task.rb
+module Maintenance
+  class UpdatePostsThrottledTask < MaintenanceTasks::Task
+    throttle_on -> { DatabaseStatus.unhealthy? }, backoff: 1.minute
+
+    def collection
+      Post.all
+    end
+
+    def count
+      collection.count
+    end
+
+    def process(post)
+      post.update!(content: "New content added on #{Time.now.utc}")
+    end
+  end
+end
+```
+
+Optionally, a `backoff` can be supplied. The default is 30 seconds.
+Note that it's up to you to define a throttling condition that makes sense for
+your app. Shopify implements `DatabaseStatus.healthy?` to check various MySQL
+metrics such as replication lag, DB threads, whether DB writes are available, 
+etc.
+
+Tasks can define multiple throttle conditions. Throttle conditions are inherited
+by descendants, meaning new conditions will be appended, rather than clobbering
```

Comment:
The present participle on `clobbering` is weird. Maybe we could reword to avoid it? `and won't remove existing conditions`?

Reply:
SGTM! I had trouble wording this section in a way that made sense, thanks! 👍